src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -342,15 +338,64 @@ private static ImmutableList<ItemData>.Builder ComputeItems(LazyItemList lazyIte
 
                 ImmutableHashSet<string> currentGlobsToIgnore = globsToIgnoreStack == null ? globsToIgnore : globsToIgnoreStack.Peek();
 
+                Dictionary<string, UpdateOperation> itemsWithNoWildcards = new Dictionary<string, UpdateOperation>(FileUtilities.GetIsFileSystemCaseSensitive() ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
+                bool addedToBatch = false;
+
                 //  Walk back down the stack of item lists applying operations
                 while (itemListStack.Count > 0)
                 {
                     var currentList = itemListStack.Pop();
 
+                    if (currentList._memoizedOperation.Operation is UpdateOperation op)
+                    {
+                        bool addToBatch = true;
+                        int i;
+                        for (i = 0; i < op.ISpec.Fragments.Count; i++)
```

Comment:
Are these fragments coming from multiple different elements, or a single one? The way I read it this applies to like `<I Update="a;b;c*;d" />` so I don't see how it applies to `<I Update="a" /><I Update="b" /><I Update="c*" /><I Update="d" />`. Misunderstanding on my part?

Reply:
No, you're right. It applies to `<I Update="a;b;c*;d" />`, but then if there aren't any globs, it doesn't do anything else, just goes to the next operation. If the next operation also happens to be an Update and not have any globs, then the two are batched together in the dictionary. For your second case, this loop would run once for `a`, then run for `b`, then try to run for `c*` but realize it has a glob, quickly do the update for (a and b), do the update for c*, then proceed with d, possibly doing the update immediately or including it in a future batch depending on if there are other Updates without wildcards after it.